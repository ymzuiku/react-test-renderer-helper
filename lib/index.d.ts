import { ReactElement, ReactType } from 'react';

interface ReactTestRendererJSON {
  type: string;
  props: { [propName: string]: any };
  children: null | ReactTestRendererJSON[];
}
interface ReactTestRendererTree extends ReactTestRendererJSON {
  nodeType: 'component' | 'host';
  instance: any;
  rendered: null | ReactTestRendererTree;
}
interface ReactTestInstance {
  instance: any;
  type: ReactType;
  props: { [propName: string]: any };
  parent: null | ReactTestInstance;
  children: Array<ReactTestInstance | string>;

  find(predicate: (node: ReactTestInstance) => boolean): ReactTestInstance;
  findByType(type: ReactType): ReactTestInstance;
  findByProps(props: { [propName: string]: any }): ReactTestInstance;

  findAll(predicate: (node: ReactTestInstance) => boolean, options?: { deep: boolean }): ReactTestInstance[];
  findAllByType(type: ReactType, options?: { deep: boolean }): ReactTestInstance[];
  findAllByProps(props: { [propName: string]: any }, options?: { deep: boolean }): ReactTestInstance[];
}

interface ReactTestRenderer {
  toJSON(): null | ReactTestRendererJSON;
  toTree(): null | ReactTestRendererTree;
  unmount(nextElement?: ReactElement<any>): void;
  update(nextElement: ReactElement<any>): void;
  getInstance(): null | ReactTestInstance;
  root: ReactTestInstance;
}
interface ISnapshot {
  status: any;
  state: any;
  props: { [propName: string]: any };
  zip: Object;
}

export function find(component: ReactTestRenderer, id: string): ReactTestInstance;
export function getStatus(key: any): any;
export function clearStatus(key: any): void;
export function toSnapshotByHooks(component: ReactTestRenderer): ISnapshot;
export function toSnapshot(component: ReactTestRenderer, ...any): ISnapshot;
