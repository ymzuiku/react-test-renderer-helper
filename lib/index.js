!function(e,t){if("object"===typeof exports&&"object"===typeof module)module.exports=t();else if("function"===typeof define&&define.amd)define([],t);else{var r=t();for(var n in r)("object"===typeof exports?exports:e)[n]=r[n]}}(window,function(){return function(e){var t={};function r(n){if(t[n])return t[n].exports;var o=t[n]={i:n,l:!1,exports:{}};return e[n].call(o.exports,o,o.exports,r),o.l=!0,o.exports}return r.m=e,r.c=t,r.d=function(e,t,n){r.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:n})},r.r=function(e){"undefined"!==typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},r.t=function(e,t){if(1&t&&(e=r(e)),8&t)return e;if(4&t&&"object"===typeof e&&e&&e.__esModule)return e;var n=Object.create(null);if(r.r(n),Object.defineProperty(n,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var o in e)r.d(n,o,function(t){return e[t]}.bind(null,o));return n},r.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return r.d(t,"a",t),t},r.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},r.p="/ymzuiku/react-test-renderer-helper.git/",r(r.s=2)}([function(e,t,r){"use strict";t.__esModule=!0,window.__testerCache={},window.__testerIsTest=!0;var n="tester__defaultKey";var o={setStatus:function(e,t){var r,o;window.__testerIsTest&&("undefined"===typeof t?(o=n,r=e):(o=e,r=t),window.__testerCache[o]=r)},getStatus:function(e){return"undefined"===typeof e&&(e=n),window.__testerCache[e]},clearStatus:function(e){"undefined"===typeof e&&(e=n),"*"===e?window.__testerCache={}:delete window.__testerCache[e]},status:window.__testerCache};t.default=o,e.exports=t.default},function(e,t){e.exports=require("react-test-renderer")},function(e,t,r){"use strict";function n(e,t,r){return t in e?Object.defineProperty(e,t,{value:r,enumerable:!0,configurable:!0,writable:!0}):e[t]=r,e}r.r(t);var o=r(1),u=r.n(o),i=r(0),f=r.n(i);r.d(t,"fixId",function(){return s}),r.d(t,"find",function(){return c}),r.d(t,"getStatus",function(){return p}),r.d(t,"clearStatus",function(){return l}),r.d(t,"toSnapshotByHooks",function(){return d}),r.d(t,"toSnapShot",function(){return y}),window.__testerIsTest=!0;var a=u.a.create().toJSON();function s(e){var t=arguments.length>1&&void 0!==arguments[1]?arguments[1]:"";if(e&&e.props){if(t.search(/\=/)>0){var r=t.split("=");return e.props[r[0]]===r[1]}if(0===t.search(/\#/)){var n=t.split("#");return e.props.id===n[1]}if(0===t.search(/\@/)){var o=t.split("@");return e.props.name===o[1]}if(0===t.search(/\./)){var u=t.split(".");return e.props.className===u[1]}return e.type===t}return!1}function c(e,t){var r,n,o=arguments.length>2&&void 0!==arguments[2]?arguments[2]:0,u=-1;return s(n=e&&e.toJSON?e.toJSON():e,t)&&(u+=1),u===o?a||n:(function e(n){if(n&&n.children)for(var i=0;i<n.children.length&&!r;i++){var f=n.children[i];s(f,t)&&(u+=1),u===o?r=f:e(f)}}(n),a||r)}function p(e){return f.a.getStatus(e)}function l(e){f.a.clearStatus(e)}function d(e,t){return!1===t?{props:e.root.props,zip:e.toJSON()}:{status:p(t),props:e.root.props,zip:e.toJSON()}}function y(e){for(var t=e.getInstance(),r={status:p(e),props:t.props,state:t.state,zip:e.toJSON()},o=arguments.length,u=new Array(o>1?o-1:0),i=1;i<o;i++)u[i-1]=arguments[i];if(u.length>0){for(var f={},a=0;a<u.length;a++)f[u[a]]=t[u[a]];return function(e){for(var t=1;t<arguments.length;t++){var r=null!=arguments[t]?arguments[t]:{},o=Object.keys(r);"function"===typeof Object.getOwnPropertySymbols&&(o=o.concat(Object.getOwnPropertySymbols(r).filter(function(e){return Object.getOwnPropertyDescriptor(r,e).enumerable}))),o.forEach(function(t){n(e,t,r[t])})}return e}({},r,{other:f})}return r}a=null}])});
//# sourceMappingURL=index.js.map