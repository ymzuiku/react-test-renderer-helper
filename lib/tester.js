!function(e,t){if("object"===typeof exports&&"object"===typeof module)module.exports=t();else if("function"===typeof define&&define.amd)define([],t);else{var n=t();for(var r in n)("object"===typeof exports?exports:e)[r]=n[r]}}(window,function(){return function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:!1,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},n.r=function(e){"undefined"!==typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&"object"===typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,"a",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p="/ymzuiku/react-test-renderer-helper.git/",n(n.s=0)}([function(e,t,n){"use strict";t.__esModule=!0,window.__testerCache={},window.__testerIsTest=!0;var r="tester__defaultKey";var o={setStatus:function(e,t){var n,o;window.__testerIsTest&&("undefined"===typeof t?(o=r,n=e):(o=e,n=t),window.__testerCache[o]=n)},getStatus:function(e){return"undefined"===typeof e&&(e=r),window.__testerCache[e]},clearStatus:function(e){"undefined"===typeof e&&(e=r),"*"===e?window.__testerCache={}:delete window.__testerCache[e]},status:window.__testerCache};t.default=o,e.exports=t.default}])});
//# sourceMappingURL=tester.js.map