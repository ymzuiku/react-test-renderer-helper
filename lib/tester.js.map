{"version":3,"sources":["../webpack/universalModuleDefinition","../webpack/bootstrap","lib/tester.lib.js"],"names":["root","factory","exports","module","define","amd","a","i","window","installedModules","__webpack_require__","moduleId","l","modules","call","m","c","d","name","getter","o","Object","defineProperty","enumerable","get","r","Symbol","toStringTag","value","t","mode","__esModule","ns","create","key","bind","n","object","property","prototype","hasOwnProperty","p","s","__testerCache","__testerIsTest","tester__defaultKey","tester__default","setStatus","b","obj","getStatus","clearStatus","status","default"],"mappings":"CAAA,SAAAA,EAAAC,GACA,qBAAAC,SAAA,kBAAAC,OACAA,OAAAD,QAAAD,SACA,uBAAAG,eAAAC,IACAD,OAAA,GAAAH,OACA,CACA,IAAAK,EAAAL,IACA,QAAAM,KAAAD,GAAA,kBAAAJ,gBAAAF,GAAAO,GAAAD,EAAAC,IAPA,CASCC,OAAA,WACD,mBCTA,IAAAC,EAAA,GAGA,SAAAC,EAAAC,GAGA,GAAAF,EAAAE,GACA,OAAAF,EAAAE,GAAAT,QAGA,IAAAC,EAAAM,EAAAE,GAAA,CACAJ,EAAAI,EACAC,GAAA,EACAV,QAAA,IAUA,OANAW,EAAAF,GAAAG,KAAAX,EAAAD,QAAAC,IAAAD,QAAAQ,GAGAP,EAAAS,GAAA,EAGAT,EAAAD,QA0DA,OArDAQ,EAAAK,EAAAF,EAGAH,EAAAM,EAAAP,EAGAC,EAAAO,EAAA,SAAAf,EAAAgB,EAAAC,GACAT,EAAAU,EAAAlB,EAAAgB,IACAG,OAAAC,eAAApB,EAAAgB,EAAA,CAA0CK,YAAA,EAAAC,IAAAL,KAK1CT,EAAAe,EAAA,SAAAvB,GACA,qBAAAwB,eAAAC,aACAN,OAAAC,eAAApB,EAAAwB,OAAAC,YAAA,CAAwDC,MAAA,WAExDP,OAAAC,eAAApB,EAAA,cAAiD0B,OAAA,KAQjDlB,EAAAmB,EAAA,SAAAD,EAAAE,GAEA,GADA,EAAAA,IAAAF,EAAAlB,EAAAkB,IACA,EAAAE,EAAA,OAAAF,EACA,KAAAE,GAAA,kBAAAF,QAAAG,WAAA,OAAAH,EACA,IAAAI,EAAAX,OAAAY,OAAA,MAGA,GAFAvB,EAAAe,EAAAO,GACAX,OAAAC,eAAAU,EAAA,WAAyCT,YAAA,EAAAK,UACzC,EAAAE,GAAA,iBAAAF,EAAA,QAAAM,KAAAN,EAAAlB,EAAAO,EAAAe,EAAAE,EAAA,SAAAA,GAAgH,OAAAN,EAAAM,IAAqBC,KAAA,KAAAD,IACrI,OAAAF,GAIAtB,EAAA0B,EAAA,SAAAjC,GACA,IAAAgB,EAAAhB,KAAA4B,WACA,WAA2B,OAAA5B,EAAA,SAC3B,WAAiC,OAAAA,GAEjC,OADAO,EAAAO,EAAAE,EAAA,IAAAA,GACAA,GAIAT,EAAAU,EAAA,SAAAiB,EAAAC,GAAsD,OAAAjB,OAAAkB,UAAAC,eAAA1B,KAAAuB,EAAAC,IAGtD5B,EAAA+B,EAAA,2CAIA/B,IAAAgC,EAAA,kCChFAxC,EAAQ6B,YAAa,EAErBvB,OAAOmC,cAAgB,GACvBnC,OAAOoC,gBAAiB,EAExB,IAAIC,EAAqB,qBA+BzB,IAAIC,EAAkB,CACpBC,UA/BF,SAA2BzC,EAAG0C,GAI5B,IAAIC,EAAKf,EAHJ1B,OAAOoC,iBAIK,qBAANI,GACTd,EAAMW,EACNI,EAAM3C,IAEN4B,EAAM5B,EACN2C,EAAMD,GAERxC,OAAOmC,cAAcT,GAAOe,IAoB5BC,UAlBF,SAA2BhB,GAIzB,MAHmB,qBAARA,IACTA,EAAMW,GAEDrC,OAAOmC,cAAcT,IAe5BiB,YAbF,SAA6BjB,GACR,qBAARA,IACTA,EAAMW,GAEI,MAARX,EACF1B,OAAOmC,cAAgB,UAEhBnC,OAAOmC,cAAcT,IAO9BkB,OAAQ5C,OAAOmC,eAGjBzC,EAAQmD,QAAUP,EAClB3C,EAAOD,QAAUA,EAAO","file":"tester.js","sourcesContent":["(function webpackUniversalModuleDefinition(root, factory) {\n\tif(typeof exports === 'object' && typeof module === 'object')\n\t\tmodule.exports = factory();\n\telse if(typeof define === 'function' && define.amd)\n\t\tdefine([], factory);\n\telse {\n\t\tvar a = factory();\n\t\tfor(var i in a) (typeof exports === 'object' ? exports : root)[i] = a[i];\n\t}\n})(window, function() {\nreturn "," \t// The module cache\n \tvar installedModules = {};\n\n \t// The require function\n \tfunction __webpack_require__(moduleId) {\n\n \t\t// Check if module is in cache\n \t\tif(installedModules[moduleId]) {\n \t\t\treturn installedModules[moduleId].exports;\n \t\t}\n \t\t// Create a new module (and put it into the cache)\n \t\tvar module = installedModules[moduleId] = {\n \t\t\ti: moduleId,\n \t\t\tl: false,\n \t\t\texports: {}\n \t\t};\n\n \t\t// Execute the module function\n \t\tmodules[moduleId].call(module.exports, module, module.exports, __webpack_require__);\n\n \t\t// Flag the module as loaded\n \t\tmodule.l = true;\n\n \t\t// Return the exports of the module\n \t\treturn module.exports;\n \t}\n\n\n \t// expose the modules object (__webpack_modules__)\n \t__webpack_require__.m = modules;\n\n \t// expose the module cache\n \t__webpack_require__.c = installedModules;\n\n \t// define getter function for harmony exports\n \t__webpack_require__.d = function(exports, name, getter) {\n \t\tif(!__webpack_require__.o(exports, name)) {\n \t\t\tObject.defineProperty(exports, name, { enumerable: true, get: getter });\n \t\t}\n \t};\n\n \t// define __esModule on exports\n \t__webpack_require__.r = function(exports) {\n \t\tif(typeof Symbol !== 'undefined' && Symbol.toStringTag) {\n \t\t\tObject.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });\n \t\t}\n \t\tObject.defineProperty(exports, '__esModule', { value: true });\n \t};\n\n \t// create a fake namespace object\n \t// mode & 1: value is a module id, require it\n \t// mode & 2: merge all properties of value into the ns\n \t// mode & 4: return value when already ns object\n \t// mode & 8|1: behave like require\n \t__webpack_require__.t = function(value, mode) {\n \t\tif(mode & 1) value = __webpack_require__(value);\n \t\tif(mode & 8) return value;\n \t\tif((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;\n \t\tvar ns = Object.create(null);\n \t\t__webpack_require__.r(ns);\n \t\tObject.defineProperty(ns, 'default', { enumerable: true, value: value });\n \t\tif(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));\n \t\treturn ns;\n \t};\n\n \t// getDefaultExport function for compatibility with non-harmony modules\n \t__webpack_require__.n = function(module) {\n \t\tvar getter = module && module.__esModule ?\n \t\t\tfunction getDefault() { return module['default']; } :\n \t\t\tfunction getModuleExports() { return module; };\n \t\t__webpack_require__.d(getter, 'a', getter);\n \t\treturn getter;\n \t};\n\n \t// Object.prototype.hasOwnProperty.call\n \t__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };\n\n \t// __webpack_public_path__\n \t__webpack_require__.p = \"/ymzuiku/react-test-renderer-helper.git/\";\n\n\n \t// Load entry module and return exports\n \treturn __webpack_require__(__webpack_require__.s = 0);\n","'use strict';\n\nexports.__esModule = true;\n\nwindow.__testerCache = {};\nwindow.__testerIsTest = true;\n\nvar tester__defaultKey = 'tester__defaultKey';\nfunction tester__setStatus(a, b) {\n  if (!window.__testerIsTest) {\n    return;\n  }\n  var obj, key;\n  if (typeof b === 'undefined') {\n    key = tester__defaultKey;\n    obj = a;\n  } else {\n    key = a;\n    obj = b;\n  }\n  window.__testerCache[key] = obj;\n}\nfunction tester__getStatus(key) {\n  if (typeof key === 'undefined') {\n    key = tester__defaultKey;\n  }\n  return window.__testerCache[key];\n}\nfunction tester__clearStatus(key) {\n  if (typeof key === 'undefined') {\n    key = tester__defaultKey;\n  }\n  if (key === '*') {\n    window.__testerCache = {};\n  } else {\n    delete window.__testerCache[key];\n  }\n}\nvar tester__default = {\n  setStatus: tester__setStatus,\n  getStatus: tester__getStatus,\n  clearStatus: tester__clearStatus,\n  status: window.__testerCache,\n};\n\nexports.default = tester__default;\nmodule.exports = exports['default'];\n"],"sourceRoot":""}