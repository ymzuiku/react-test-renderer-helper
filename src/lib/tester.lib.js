'use strict';

exports.__esModule = true;

window.__testerCache = {};
window.__testerIsTest = true;

var tester__defaultKey = 'tester__defaultKey';
function tester__setStatus(a, b) {
  if (typeof window === 'undefined' || !window.__testerIsTest) {
    return;
  }
  var obj, key;
  if (typeof b === 'undefined') {
    key = tester__defaultKey;
    obj = a;
  } else {
    key = a;
    obj = b;
  }
  window.__testerCache[key] = obj;
}
function tester__getStatus(key) {
  if (typeof key === 'undefined') {
    key = tester__defaultKey;
  }
  return window.__testerCache[key];
}
function tester__clearStatus(key) {
  if (typeof key === 'undefined') {
    key = tester__defaultKey;
  }
  if (key === '*') {
    window.__testerCache = {};
    tester__default.status = window.__testerCache;
  } else {
    delete window.__testerCache[key];
    tester__default.status = window.__testerCache;
  }
}
var tester__default = {
  setStatus: tester__setStatus,
  getStatus: tester__getStatus,
  clearStatus: tester__clearStatus,
  status: window.__testerCache,
};

exports.default = tester__default;
module.exports = exports['default'];
