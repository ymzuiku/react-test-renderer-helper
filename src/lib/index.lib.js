import renderer from 'react-test-renderer';
import tester from './tester.lib';
window.__testerIsTest = true;

let rendererType = renderer.create().toJSON();
rendererType = null;

export function fixId(ele, id = '') {
  if (ele && ele.props) {
    if (id.search(/\=/) > 0) {
      const kv = id.split('=');
      return ele.props[kv[0]] === kv[1];
    }
    if (id.search(/\#/) === 0) {
      const kv = id.split('#');
      return ele.props.id === kv[1];
    }
    if (id.search(/\@/) === 0) {
      const kv = id.split('@');
      return ele.props.name === kv[1];
    }
    if (id.search(/\./) === 0) {
      const kv = id.split('.');
      return ele.props.className === kv[1];
    }
    return ele.type === id;
  }
  return false;
}

export function find(component, id, num = 0) {
  let child;
  let jumpNum = -1;
  function _getChild(comp) {
    if (comp && comp.children) {
      for (let i = 0; i < comp.children.length; i++) {
        if (child) {
          break;
        }
        const ele = comp.children[i];
        if (fixId(ele, id)) {
          jumpNum += 1;
        }
        if (jumpNum === num) {
          child = ele;
        } else {
          _getChild(ele);
        }
      }
    }
  }
  // 如果传入的 component 没有 toJSON，帮它toJSON
  let rootComp;
  if (component && component.toJSON) {
    rootComp = component.toJSON();
  } else {
    rootComp = component;
  }
  // 首先匹配根节点
  if (fixId(rootComp, id)) {
    jumpNum += 1;
  }
  if (jumpNum === num) {
    return rendererType || rootComp;
  }
  // 如果根节点没有匹配，递归子节点
  _getChild(rootComp);
  return rendererType || child;
}

export function getStatus(key) {
  return tester.getStatus(key);
}

export function clearStatus(key) {
  tester.clearStatus(key);
}

export function toSnapshotByHooks(component, id) {
  if (id === false) {
    return {
      props: component.root.props,
      zip: component.toJSON(),
    };
  }
  return {
    status: getStatus(id),
    props: component.root.props,
    zip: component.toJSON(),
  };
}

export function toSnapshot(component, params) {
  const instance = component.getInstance();
  const returnEle = {
    status: getStatus(component),
    props: instance.props,
    state: instance.state,
    zip: component.toJSON(),
  };
  if (params && params.length > 0) {
    let other = {};
    for (let i = 0; i < params.length; i++) {
      other[params[i]] = instance[params[i]];
    }
    return {
      ...returnEle,
      other,
    };
  }
  return returnEle;
}
