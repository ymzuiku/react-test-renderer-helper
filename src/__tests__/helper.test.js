import { find, getStatus, clearStatus, toSnapshotByHooks, toSnapshot, fixId } from '../lib/index.lib';
import tester from '../lib/tester.lib';
import React from 'react';
import renderer from 'react-test-renderer';

class App extends React.Component {
  data = {
    name: 'dipu',
    age: 100,
  };
  render() {
    return (
      <div>
        <div id="test-id">test-id</div>
        <div name="test-name">test-name</div>
        <div className="test-className">test-className</div>
        <img alt="test-alt" src="test-alt" />
        <input name="test-type0" />
        <input name="test-type1" />
        <input name="test-type2" />
      </div>
    );
  }
}

describe('find', () => {
  test('find key', () => {
    const comp = renderer.create(<App />);
    expect(find(comp, 'id=test-id')).toMatchSnapshot('find-id');
    expect(find(comp, 'name=test-name')).toMatchSnapshot('find-name');
    expect(find(comp, 'className=test-className')).toMatchSnapshot('find-className');
    expect(find(comp, 'alt=test-alt')).toMatchSnapshot('find-alt');

    expect(find(comp, 'input', 1)).toMatchSnapshot('find-type');
  });
  test('find root', () => {
    const comp = renderer.create(<App />);
    expect(find(comp, 'div')).toMatchSnapshot();
  });
  test('find id', () => {
    const comp = renderer.create(<App />);
    expect(find(comp, '#test-id')).toMatchSnapshot();
    expect(find(comp, '#test-id')).toEqual(find(comp, 'id=test-id'));
  });
  test('find name', () => {
    const comp = renderer.create(<App />);
    expect(find(comp, '@test-name')).toMatchSnapshot();
    expect(find(comp, '@test-name')).toEqual(find(comp, 'name=test-name'));
  });
  test('find className', () => {
    const comp = renderer.create(<App />);
    expect(find(comp, '.test-className')).toMatchSnapshot();
    expect(find(comp, '.test-className')).toEqual(find(comp, 'className=test-className'));
  });
});

describe('getStatus', () => {
  test('getStatus-set-default-key', () => {
    tester.setStatus({ dog: 'default', age: 10 });
    expect(getStatus()).toEqual({ dog: 'default', age: 10 });
  });
  test('getStatus-set-user-key', () => {
    tester.setStatus('user', { dog: 'user', age: 10 });
    expect(find());
    expect(getStatus('user')).toEqual({ dog: 'user', age: 10 });
  });
});

describe('clearStatus', () => {
  test('clearStatus-default-key', () => {
    tester.setStatus({ dog: 'default', age: 10 });
    clearStatus();
    expect(getStatus()).toEqual(undefined);
  });
  test('clearStatus-user-key', () => {
    tester.setStatus('user', { dog: 'default', age: 10 });
    clearStatus('user');
    expect(getStatus('user')).toEqual(undefined);
  });
  test('clearStatus-*-key', () => {
    tester.setStatus('user', { dog: 'default', age: 10 });
    tester.setStatus('other', { dog: 'default', age: 10 });
    clearStatus('*');
    expect(tester.status).toEqual({});
  });
});

describe('toSnapshot', () => {
  test('toSnapshot', () => {
    const comp = renderer.create(<App />);
    expect(toSnapshot(comp)).toMatchSnapshot('toSnapshot');
  });

  test('toSnapshot', () => {
    const comp = renderer.create(<App />);
    expect(toSnapshot(comp, ['data'])).toMatchSnapshot('toSnapshot');
  });
});

describe('toSnapshotByHooks', () => {
  test('toSnapshotByHooks', () => {
    const comp = renderer.create(<App />);
    expect(toSnapshotByHooks(comp)).toMatchSnapshot('toSnapshot');
  });
  test('toSnapshotByHooks no-status', () => {
    const comp = renderer.create(<App />);
    expect(toSnapshotByHooks(comp, false)).toMatchSnapshot('toSnapshot-no-status');
  });
});

describe('fixId', () => {
  test('fixId', () => {
    const comp = renderer.create(<App />);
    const input = find(comp, 'input', 2);
    expect(fixId(input, 'name=test-type2')).toEqual(true);
  });
});

describe('tester', () => {
  test('no-window.__testerIsTest', () => {
    window.__testerIsTest = false;
    expect(tester.setStatus({ test: 'noTest' })).toEqual(undefined);
  });
});
