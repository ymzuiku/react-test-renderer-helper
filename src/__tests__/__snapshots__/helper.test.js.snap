// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`find find className 1`] = `
<div
  className="test-className"
>
  test-className
</div>
`;

exports[`find find id 1`] = `
<div
  id="test-id"
>
  test-id
</div>
`;

exports[`find find key: find-alt 1`] = `
<img
  alt="test-alt"
  src="test-alt"
/>
`;

exports[`find find key: find-className 1`] = `
<div
  className="test-className"
>
  test-className
</div>
`;

exports[`find find key: find-id 1`] = `
<div
  id="test-id"
>
  test-id
</div>
`;

exports[`find find key: find-name 1`] = `
<div
  name="test-name"
>
  test-name
</div>
`;

exports[`find find key: find-type 1`] = `
<input
  name="test-type1"
/>
`;

exports[`find find name 1`] = `
<div
  name="test-name"
>
  test-name
</div>
`;

exports[`find find root 1`] = `
<div>
  <div
    id="test-id"
  >
    test-id
  </div>
  <div
    name="test-name"
  >
    test-name
  </div>
  <div
    className="test-className"
  >
    test-className
  </div>
  <img
    alt="test-alt"
    src="test-alt"
  />
  <input
    name="test-type0"
  />
  <input
    name="test-type1"
  />
  <input
    name="test-type2"
  />
</div>
`;

exports[`toSnapshot toSnapshot: toSnapshot 1`] = `
Object {
  "props": Object {},
  "state": null,
  "status": undefined,
  "zip": <div>
    <div
      id="test-id"
    >
      test-id
    </div>
    <div
      name="test-name"
    >
      test-name
    </div>
    <div
      className="test-className"
    >
      test-className
    </div>
    <img
      alt="test-alt"
      src="test-alt"
    />
    <input
      name="test-type0"
    />
    <input
      name="test-type1"
    />
    <input
      name="test-type2"
    />
  </div>,
}
`;

exports[`toSnapshot toSnapshot: toSnapshot 2`] = `
Object {
  "other": Object {
    "data": Object {
      "age": 100,
      "name": "dipu",
    },
  },
  "props": Object {},
  "state": null,
  "status": undefined,
  "zip": <div>
    <div
      id="test-id"
    >
      test-id
    </div>
    <div
      name="test-name"
    >
      test-name
    </div>
    <div
      className="test-className"
    >
      test-className
    </div>
    <img
      alt="test-alt"
      src="test-alt"
    />
    <input
      name="test-type0"
    />
    <input
      name="test-type1"
    />
    <input
      name="test-type2"
    />
  </div>,
}
`;

exports[`toSnapshotByHooks toSnapshotByHooks no-status: toSnapshot-no-status 1`] = `
Object {
  "props": Object {},
  "zip": <div>
    <div
      id="test-id"
    >
      test-id
    </div>
    <div
      name="test-name"
    >
      test-name
    </div>
    <div
      className="test-className"
    >
      test-className
    </div>
    <img
      alt="test-alt"
      src="test-alt"
    />
    <input
      name="test-type0"
    />
    <input
      name="test-type1"
    />
    <input
      name="test-type2"
    />
  </div>,
}
`;

exports[`toSnapshotByHooks toSnapshotByHooks: toSnapshot 1`] = `
Object {
  "props": Object {},
  "status": undefined,
  "zip": <div>
    <div
      id="test-id"
    >
      test-id
    </div>
    <div
      name="test-name"
    >
      test-name
    </div>
    <div
      className="test-className"
    >
      test-className
    </div>
    <img
      alt="test-alt"
      src="test-alt"
    />
    <input
      name="test-type0"
    />
    <input
      name="test-type1"
    />
    <input
      name="test-type2"
    />
  </div>,
}
`;
